### PR TITLE
[Soroban] Add scValByType function

### PR DESCRIPTION
**Pull request by CassioMG**
https://stellarorg.atlassian.net/jira/software/c/projects/WAL/boards/37?selectedIssue=WAL-1368

### CODE REVIEW

**Review by piyalbasu — APPROVED** [review]

File: src/walletSdk/Utils/soroban.ts [hunk 1]
```diff
@@ -165,3 +165,83 @@ export const parseTokenAmount = (
 
   return BigInt(parsed.toString());
 };
+
+// This function attempt to convert soroban contract values
+// to common types like string, array, buffer, JSON string, etc.
+export const scValByType = (scVal: xdr.ScVal) => {
```
Comment (Ifropc) [inline src/walletSdk/Utils/soroban.ts:171]: We probably should add module for Soroban in the future, considering that the codebase should grow for it 🤔

**Review by Ifropc — APPROVED** [review]

File: src/walletSdk/Utils/soroban.ts [hunk 2]
```diff
@@ -165,3 +165,83 @@ export const parseTokenAmount = (
 
   return BigInt(parsed.toString());
 };
+
+// This function attempt to convert soroban contract values
```
Comment (acharb) [inline src/walletSdk/Utils/soroban.ts:169]: can we use the jsdoc comment style for describing what the method's doing? [for example ](https://github.com/stellar/typescript-wallet-sdk/blob/e9943d82d28e5a6206c5636a4e53e93b4dc3e8f1/%40stellar/typescript-wallet-sdk/src/walletSdk/Anchor/Sep6.ts#L98-L115)

**Review by acharb — APPROVED** [review]
had 1 comment otherwise lgtm

File: src/walletSdk/Utils/soroban.ts (hunk 2 above)
Comment (acharb) [reply src/walletSdk/Utils/soroban.ts:169]: and actually if we could do for all of the methods in this file 🙏 

will keep it consistent with other code in the codebase, and helps users understand how to use the methods

Comment (CassioMG) [reply src/walletSdk/Utils/soroban.ts:169]: for sure!

**Comment by CassioMG** [comment]
Closing this in favor of https://github.com/stellar/typescript-wallet-sdk/pull/121 which is on the new soroban workspace

**Pull request closed without merging**